structuralcodes/codes/ec2_2004/annex_b_shrink_and_creep.py

```diff
@@ -0,0 +1,184 @@
+        t_dead_load (int): the number of days before the dead load is applied, defaults to 7 days,
+        t_live_load (int): the number of days before the live load is applied, defaults to 28 days.
+
+    Returns:
+        list[2]: a vector containing the two creep values for the dead and live load, respectively.
+
+    Raises:
+        ValueError: checks if the cement class equals R, N or S.
+    """
+    f_cm = fck + 8
+
+    # The cement class will decide the value of the three constants alpha_cement, ds1 and ds2.
+    # The values are given in (B.9) and (B.12)
+    if cement_class == 'R':
+        alpha_cement = 1
+    elif cement_class == 'N':
+        alpha_cement = 0
+    elif cement_class == 'S':
+        alpha_cement = -1
+    else:
+        raise ValueError(f'cement_class={cement_class}, expected R, N or S')
+
+    beta_H, phi_RH = calculate_phiRH_betaH(h_0, f_cm, RH)
+    beta_fcm = 16.8 / f_cm**0.5  # (B.4)
+
+    # Collects and readies the time variables for the applied dead and live load
+    t_forever = 18262  # OVE SLETTEN AS software defaults to 25 years, the support doesn't know why
+    t0 = [t_dead_load, t_live_load]
+
+    # This part calculates creep for each of the two load cases
+    phi_t_t0 = [0, 0]  # just an initialization
+    for i in range(2):
+        t0_adjusted = max(
+            t0[i] * (9 / (2 + t0[i] ** 1.2) + 1) ** alpha_cement, 0.5
+        )  # (B.9)
+        beta_t0 = 1 / (0.1 + t0_adjusted**0.20)  # (B.5)
+        beta_c = (
+            (t_forever - t0[i]) / (beta_H + t_forever - t0[i])
+        ) ** 0.3  # (B.7)
+        phi_0 = phi_RH * beta_fcm * beta_t0  # (B.2)
+        phi_t_t0[i] = phi_0 * beta_c  # (B.1)
+    return phi_t_t0
+
+
+def calculate_phiRH_betaH(h_0, f_cm, RH):
+    """This function will calculate eps_cd_0 according to Annex B in EUROCODE 1992-1-1:2004.
+
+    Args:
+        h_0 (float): The cement class, defaults to 'R'. Possible values: 'R', 'N', 'S',
+        f_cm (float): The characteristic concrete strength,
+        RH (float): Calculation parameter from equation (B.12).
+
+    Returns:
+        (float, float): the calculation parameters defined in (B.3) and (B.8)
+
+    """
+    # (B.8c) Alpha 1 to 3 is a constant where only f_cm is a variable
+    alpha_1, alpha_2, alpha_3 = (
+        (35 / f_cm) ** 0.7,
+        (35 / f_cm) ** 0.2,
+        (35 / f_cm) ** 0.5,
+    )  # (B.8)
+
+    if f_cm <= 35:
+        # (B.8a) and (B.3a) applies
+        beta_H = min(1.5 * (1 + (0.012 * RH) ** 18) * h_0 + 250, 1500)
+        phi_RH = 1 + (1 - RH / 100) / (0.1 * h_0 ** (1 / 3))
+    else:
+        # (B.8b) and (B.3b) applies
+        beta_H = min(
+            1.5 * (1 + (0.012 * RH) ** 18) * h_0 + 250 * alpha_3,
+            1500 * alpha_3,
+        )
+        phi_RH = (
+            1 + (1 - RH / 100) / (0.1 * h_0 ** (1 / 3)) * alpha_1
+        ) * alpha_2
+    return beta_H, phi_RH
+
+
+def annex_B_shrinkage(
```

Comment:
We should have from annex B equations for B.11 and B.12 (the next function in this module).
The logic in this function should be inside module for chapter 3 and not annex b?
Maybe consider factoring out this function?
Anyhow, again I would split all the small steps:
1. Function for 3.8 eps_cs (eps_cd, eps_ca)
2. Function for eps_cd (3.9) with inputs beta_ds, kh, eps_cd_0 (for this one the user should use the function from Annex B module).
3. Function for beta_ds(t,ts) with inputs t, ts, h0
4. funtion eps_ca with inputs beta_as and eps_ca_inf (eq. 3.11)
5. Function for eps_ca_inf with input fck (eq. 3.12)
6. Functions for beta_as with input t (eq. 3.13)

Reply:
Hi. Thanks for many good suggestions to my code. The code was written in a very specific way to yield exactly what I needed to train my AI. Traces of this are still seen and will be adressed (for instance: calculating dead and live load for 7 and 8 days to inf). I hope I will have my code ready for a new review before next week.

Reply:
I have pushed my new code. Should I now press the "ready for review" button?